### PR TITLE
feat: Add external provisioner daemons

### PR DESCRIPTION
**Pull request by kylecarbs**
This adds the concept of tags to provisioner daemons.

When creating a template or pushing a new version, `--provisioner-tag` is now a command-line option. This allows you to target provisioners with the matching set of tags.

`coder provisionerd start --tag key=value` launches a new provisioner daemon.

`scope=user` is a _special_ tag that will auto-insert the `owner=<user-id>` tag when the user initializes build operations for their resources. This solves #909 by allowing users to run a daemon, and template admins can target them.  

This feature lacks polish right now (notably, the UI will show a "Pending" state if a daemon isn't matching the specified tags), but that seems fine while we initially test. This doesn't make any irreversible changes, so it should be good to merge.

### CODE REVIEW

**Review by Kira-Pilot — APPROVED** [review]
FE ✅

**Review by coadler — APPROVED** [review]

File: coderd/database/queries.sql.go [hunk 1]
```diff
@@ -2487,19 +2494,22 @@ WHERE
...
 			1
-	) RETURNING id, created_at, updated_at, started_at, canceled_at, completed_at, error, organization_id, initiator_id, provisioner, storage_method, type, input, worker_id, file_id
+	) RETURNING id, created_at, updated_at, started_at, canceled_at, completed_at, error, organization_id, initiator_id, provisioner, storage_method, type, input, worker_id, file_id, tags
 `
 
 type AcquireProvisionerJobParams struct {
 	StartedAt sql.NullTime      `db:"started_at" json:"started_at"`
 	WorkerID  uuid.NullUUID     `db:"worker_id" json:"worker_id"`
 	Types     []ProvisionerType `db:"types" json:"types"`
+	Tags      json.RawMessage   `db:"tags" json:"tags"`
```
[the remark this answers is not in the document]
Comment (kylecarbs) [reply coderd/database/queries.sql.go:2512]: No I'm not D: D: D:

File: coderd/database/queries/provisionerjobs.sql [hunk 2]
```diff
@@ -22,6 +22,8 @@ WHERE
 			AND nested.canceled_at IS NULL
 			AND nested.completed_at IS NULL
 			AND nested.provisioner = ANY(@types :: provisioner_type [ ])
+			-- Ensure the caller satisfies all job tags.
+			AND @tags :: jsonb @> nested.tags
```
[the remark this answers is not in the document]
Comment (kylecarbs) [reply coderd/database/queries/provisionerjobs.sql:26]: Sure

File: coderd/database/migrations/000077_provisioner_daemon_tags.down.sql [hunk 3]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE provisioner_daemons DROP COLUMN auth_token;
+ALTER TABLE provisioner_daemons DROP COLUMN tags;
```
[the remark this answers is not in the document]
Comment (kylecarbs) [reply coderd/database/migrations/000077_provisioner_daemon_tags.down.sql:2]: `auth_token` isn't used since the user just authenticates it now.

**Pull request merged**